dask/dataframe/tests/test_multi.py

```diff
@@ -1030,6 +1030,59 @@ def test_merge_empty_left_df(shuffle_method, how):
     merged.map_partitions(lambda x: x, meta=merged._meta).compute()
 
 
+@pytest.mark.parametrize("how", ["inner", "outer", "left", "right"])
+@pytest.mark.parametrize("broadcast", [False, True])
+def test_merge_shuffled(how, broadcast):
+    index = ["a", "b"]
+    lhs_i = dd.from_dict(
+        {
+            "a": [1, 2, 3, 4, 5] * 20,
+            "b": [1, 2] * 50,
+            "c": range(100),
+        },
+        npartitions=5,
+    )
+    lhs = lhs_i.shuffle(index)
+
+    rhs_i = dd.from_dict(
+        {
+            "a": [1, 2, 3, 4] * 25,
+            "b": [1, 2] * 50,
+            "d": range(100),
+        },
+        npartitions=4,
+    )
+    rhs = rhs_i.shuffle(index)
+    expect = lhs_i.merge(rhs_i, on=index, how=how, broadcast=broadcast)
+
+    # Use `left_shuffled`
+    got = lhs.merge(rhs_i, on=index, how=how, left_shuffled=True, broadcast=broadcast)
+    assert_eq(got, expect, check_index=False, check_divisions=not broadcast)
```

Comment:
@randerzander @ayushdg - Do these `left_shuffled`/`right_shuffled` options seem like they would be helpful in dask-sql?

Reply:
This would definitely be helpful with perf for queries that are a sequence of joins with a common join key from a table is being used for the joins, or when once of the tables being joined on is already partitioned by the key on disk. Here's the issue I'd written up a while back to track for this on the dask-sql side https://github.com/dask-contrib/dask-sql/issues/968.

Specifically the `fast path` I talk about in that issue for joins could just be specifying `left_shuffled/right_shuffled` instead of rewriting the logic to shuffle the other table + doing a map_partitions style merge.